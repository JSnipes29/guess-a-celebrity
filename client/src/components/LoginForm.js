/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import '../assets/App.css';
import Account from '../modules/Account';
import Api from '../modules/Api';

export default function LoginForm({ setScores, setUsers }) {
  const [username, setUsername] = useState({ name: '' });
  const submit = async (e) => {
    e.preventDefault();
    const result = await Api.setup(username.name);
    if (result === -1) {
      alert('Please enter a valid name');
    } else if (result === -2) {
      alert('Please enter letters and numbers only.');
    } else {
      setUsers({
        name: username.name,
      });
      setScores({
        userMax: result.maxScore,
        globalMax: result.globalMaxScore,
        globalName: result.globalName,
      });
    }
  };
  return (
    <div>
      <div>
        <div>Name:</div>
        <input type="text" name="name" id="name" onChange={(e) => setUsername({ ...username, name: e.target.value })} value={username.name} />
      </div>
      <input id="login" className="button" type="submit" value="login" onClick={submit} />
    </div>
  );
}
