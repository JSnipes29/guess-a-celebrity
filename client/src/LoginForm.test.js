/**
 * @jest-environment jsdom
 */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import LoginForm from './components/LoginForm';

function LoginComponent() {
  const [, setUser] = useState({ name: '' });
  const [, setScores] = useState({
    userMax: 0,
    globalMax: 0,
    globalName: '',
  });
  return (
    <LoginForm setScores={setScores} setUser={setUser} />
  );
}
test('renders enter your name', () => {
  render(<LoginComponent />);
  const linkElement = screen.getByText(/Name:/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders input name', () => {
  render(<LoginComponent />);
  expect(screen.getByDisplayValue('').id).toBe('name');
});

test('snapshot test', () => {
  const component = renderer.create(<LoginComponent />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
