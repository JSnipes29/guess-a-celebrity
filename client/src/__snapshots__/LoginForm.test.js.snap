// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <div>
    <div>
      Name:
    </div>
    <input
      id="name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <input
    className="button"
    onClick={[Function]}
    type="submit"
    value="login"
  />
</div>
`;
